.github/workflows/go.yml

```diff
@@ -135,29 +135,29 @@ jobs:
                   echo "/usr/local/go/bin" >> $GITHUB_PATH
                   echo "$HOME/go/bin" >> $GITHUB_PATH
 
-            - name: Install client dependencies
+            - name: Install tools for Go ${{ matrix.go }}
               working-directory: ./go
-              run: make install-tools
+              run: make install-tools-go${{ matrix.go }}
 
             - name: Build client
               working-directory: ./go
               run: make build
 
             - name: Run linters
               working-directory: ./go
-              run: make lint
+              run: make lint-ci
 
-            - name: Run unit tests
+            - name: Run tests
               working-directory: ./go
-              run: make unit-test-report
+              run: make test-and-report
 
             - name: Upload test reports
               if: always()
               continue-on-error: true
               uses: actions/upload-artifact@v4
               with:
                   name: test-reports-go-${{ matrix.go }}-amazon-linux-latest
-                  path: go/reports/unit-test-report.html
+                  path: go/reports/test-report.html
```

Comment:
The CI changes were originally in here because the .go changes triggered new build failures that revealed a problem with the Go CI. I've now separated the CI changes out to #1055. This PR will need to be merged after the CI is fixed in #1055.